libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -10,39 +10,24 @@ import com.mapbox.navigation.base.route.NavigationRoute
 import com.mapbox.navigation.base.route.NavigationRouterCallback
 import com.mapbox.navigation.base.route.NavigationRouterRefreshCallback
 import com.mapbox.navigation.base.route.Router
-import com.mapbox.navigation.core.SetRoutes
 import com.mapbox.navigation.core.internal.utils.initialLegIndex
-import com.mapbox.navigation.core.internal.utils.mapToReason
 import java.util.concurrent.CopyOnWriteArraySet
 
 /**
  * Default implementation of [DirectionsSession].
  *
  * @property router route fetcher. Usually Onboard, Offboard or Hybrid
- * @property routes a list of [DirectionsRoute]. Fetched from [Router] or might be set manually
+ * @property routesUpdatedResult info of last routes update. Fetched from [Router] or might be set manually
  */
 internal class MapboxDirectionsSession(
     private val router: NavigationRouterV2,
 ) : DirectionsSession {
 
     private val routesObservers = CopyOnWriteArraySet<RoutesObserver>()
 
-    /**
-     * Routes that were fetched from [Router] or set manually.
-     * On [routes] change notify registered [RoutesObserver]
-     *
-     * @see [registerRoutesObserver]
-     */
-    override var routes: List<NavigationRoute> = emptyList()
-        private set(value) {
-            field = value
-            routesInitialized = true
-        }
-
-    private var routesInitialized = false
-
-    @RoutesExtra.RoutesUpdateReason
-    private var routesUpdateReason: String = RoutesExtra.ROUTES_UPDATE_REASON_CLEAN_UP
+    override var routesUpdatedResult: RoutesUpdatedResult? = null
+    override val routes: List<NavigationRoute>
```

Comment:
nit: IMHO it should be the extension like `fun RoutesUpdatedResult?.routesOrEmptyList()`, not affect the logic so up to you

Reply:
seems the current duplicates the API actually, but I'm ok with both 👍